augraphy/augmentations/delaunay.py

```diff
@@ -179,52 +162,140 @@ def apply_augmentation(self):
-                    np.random.randint(min_val, max_val),
-                )
+                colors = [
+                    (250, 235, 215),
+                    (240, 240, 230),
+                    (253, 245, 230),
+                    (255, 245, 238),
+                    (255, 248, 220),
+                    (248, 248, 255),
+                    (255, 240, 245),
+                    (245, 255, 250),
+                    (255, 250, 250),
+                    (240, 248, 255),
+                    (240, 255, 255),
+                    (240, 255, 240),
+                    (255, 245, 238),
+                    (243, 229, 171),
+                    (250, 250, 210),
+                ]
+                alt_colors = [
+                    (255, 255, 240),
+                    (255, 250, 205),
+                    (238, 232, 170),
+                    (255, 255, 224),
+                    (255, 239, 213),
+                ]
+                if (
+                    pt1[0]
+                    and pt2[0]
+                    and pt3[0] <= self.width * 0.80
+                    and (pt1[0] and pt2[0] and pt3[0] >= self.width * 0.40)
+                ):
+                    color = colors[np.random.randint(len(colors))]  # choose from colors
+
+                elif pt1[0] and pt2[0] and pt3[0] <= self.width * 0.40:
+                    color = alt_colors[np.random.randint(len(alt_colors))]
+                else:
+                    color = alt_colors[np.random.randint(len(alt_colors))]
+                color = colors[np.random.randint(len(colors))]  # choose from colors
                 cv2.fillConvexPoly(img, np.array([pt1, pt2, pt3]), color)
         return img
 
     # Applies the Augmentation to input data.
     def __call__(self, image, layer=None, force=False):
         if force or self.should_run():
+            self.width = self.height = random.choice(
+                [400, 480, 500, 600, 640, 720],
+            )  # randomly selecting the width and the height of the background pattern
+            self.n_points = random.randint(
+                self.n_points_range[0],
+                self.n_points_range[1],
+            )  # randomly selecting the number of points in the geometric plane
+            self.n_horizontal_points = random.randint(
+                self.n_horizontal_points_range[0],
+                self.n_horizontal_points_range[1],
+            )  # randomly selecting the edge horizontal points in the goemetric plane
+            self.n_vertical_points = random.randint(
+                self.n_vertical_points_range[0],
+                self.n_vertical_points_range[1],
+            )  # randonly selecting the edge vertical points in the geometric plane
+            if self.noise_type == "random":
+                self.perlin = random.choice(
+                    [True, False],
+                )  # randomly select to apply Perlin Noise on top of the Tessellation
+            else:
+                self.perlin = False
```

Comment:
I think here should be `self.perlin` = `self.noise_type` because if user set True on perlin, we need to get it here.

Reply:
Hi @kwcckw, the only option we are giving the user is to select the noise_type. It will either be None or "random" for now. If it is random, then Perlin Noise may or may not be added. It is None or any other value, Perlin Noise is not added.

Explicitly the user cannot select True for Noise_type. What I would suggest is that the user selects noise_type = "perlin" so that perlin noise is integrated. Is this ok ?

Reply:
Yes, this should work, so you need to mention this in the class descriptions too.

Reply:
Yes @kwcckw i will add this to the class description